src/libs/actions/data-lake.ts

```diff
@@ -52,7 +52,17 @@ export const getDataLakeVariableData = (id: string): string | number | boolean |
   return dataLakeVariableData[id]
 }
 
-export const setDataLakeVariableData = (id: string, data: string | number | boolean): void => {
+export const setDataLakeVariableData = (
+  id: string,
+  data: object | string | number | boolean | Array<string | number>
+): void => {
+  if (data === null) return
+  if (typeof data !== 'string' && typeof data !== 'number') return
+
+  if (dataLakeVariableInfo[id] === undefined) {
+    createDataLakeVariable(new DataLakeVariable(id, id, typeof data === 'string' ? 'string' : 'number'))
```

Comment:
Should we keep this? I feel like it creates an incentive for people not to explicitly register their variables.

With the idea of also adding unit information to the variables, this will become more and more of a problem.

Reply:
I can make it happen only in ardupilot.ts for the mavlink data. I assume we don't want to register all mavlink data manually

Reply:
Agree, but in this case, to enforce the interface and so we can improve it in the future without having to change a lot of places in the code, I think it would be better to put this "type inferring" in the variable registration and register explicitly in the ardupilot file, even if without the type. It would just be this if clause, but in the ardupilot file instead of the data-lake itself. What do you think?